File: backends/ruby-hanami/app/domain/staff/value_objects.rb
# frozen_string_literal: true
#
# スタッフの値オブジェクトを定義するドメインオブジェクトモジュール。
#
# @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>

module Domain
  module Staff
    # スタッフ一覧の表示用値オブジェクトです。
    ListItem = Struct.new(:id, :name, :email, :role, :status, :created_at, :updated_at, keyword_init: true)

    # スタッフ詳細の値オブジェクトです。
    Vo = Struct.new(:id, :name, :avatar, :role, keyword_init: true)
  end
end

File: backends/go-echo/internal/domain/staff/value_objects.go
package staff

import "time"

type ListItem struct {
	ID        uint
	Name      string
	Email     string
	Role      int
	Status    int
	CreatedAt time.Time
	UpdatedAt time.Time
}

type Vo struct {
	ID     uint
	Name   string
	Avatar *string
	Role   int
}
